bundles/org.openhab.ui/web/src/assets/definitions/blockly/blocks-http.js

```diff
@@ -0,0 +1,184 @@
+
+          const blockAfter = (this.getInput('requestHeader')) ? 'requestHeader' : (this.getInput('payload')) ? 'payload' : undefined
+
+          if (blockAfter) {
+            this.moveInputBefore('timeoutInput', blockAfter)
+          }
+          if (addNumBlock) {
+            const parentConnection = timeoutInput.connection
+            const mathNumberBlock = this.workspace.newBlock('math_number')
+            mathNumberBlock.setFieldValue('3000', 'NUM')
+            mathNumberBlock.initSvg()
+            mathNumberBlock.render()
+            parentConnection.connect(mathNumberBlock.outputConnection)
+          }
+        }
+      } else {
+        if (this.getInput('timeoutInput')) {
+          const parentConnection = this.getInput('timeoutInput').connection
+          const targetBlock = parentConnection.targetBlock()
+          if (targetBlock) {
+            targetBlock.unplug(true)
+            targetBlock.dispose(true)
+          }
+          this.removeInput('timeoutInput')
+        }
+      }
+      if (hasHeader) {
+        if (!this.getInput('requestHeader')) {
+          const headerInput = this.appendValueInput('requestHeader')
+            .setCheck('Dictionary')
+            .appendField('with Headers')
+          const blockAfter = (this.getInput('payload')) ? 'payload' : undefined
+
+          if (blockAfter) {
+            this.moveInputBefore('requestHeader', blockAfter)
+          }
+        }
+      } else {
+        if (this.getInput('requestHeader')) {
+          this.removeInput('requestHeader')
+        }
+      }
+    },
+    onClickTimeout () {
+      let block = this.getSourceBlock()
+      block.hasTimeout = !block.hasTimeout
+      block.updateShape(block.hasTimeout, true, block.hasHeader)
+    },
+    onClickHeader () {
+      let block = this.getSourceBlock()
+      block.hasHeader = !block.hasHeader
+      block.updateShape(block.hasTimeout, true, block.hasHeader)
+    },
+    mutationToDom: function () {
+      let container = Blockly.utils.xml.createElement('mutation')
+      container.setAttribute('hasTimeout', this.hasTimeout)
+      container.setAttribute('hasHeader', this.hasHeader)
+      return container
+    },
+    domToMutation: function (xmlElement) {
+      let hasTimeout = xmlElement.getAttribute('hasTimeout') === 'true'
+      let hasHeader = xmlElement.getAttribute('hasHeader') === 'true'
+      this.updateShape(hasTimeout, false, hasHeader)
+    }
+  }
+
+  /**
+   * Implements GET, DELETE, POST and PUT requests
+   *
+   * @param block
+   * @returns {[string,number]}
+   */
+  javascriptGenerator.forBlock['oh_httprequest'] = function (block) {
+    const requestType = block.getFieldValue('requestType')
+    const contentType = block.getFieldValue('contentType')
+    const headers = javascriptGenerator.valueToCode(block, 'requestHeader', javascriptGenerator.ORDER_ATOMIC)
+
+    let timeout = javascriptGenerator.valueToCode(block, 'timeoutInput', javascriptGenerator.ORDER_ATOMIC)
```

Comment:
```suggestion
    const timeout = javascriptGenerator.valueToCode(block, 'timeoutInput', javascriptGenerator.ORDER_ATOMIC) || 3000
```

... and remove the next if statement.

Reply:
Thanks: good hint - I didn't know that syntax 😍

Reply:
JS has some very nice syntactic features, but probably too many to know them all.